internal/ethapi/api.go

```diff
@@ -460,6 +462,30 @@ func (s *PrivateAccountAPI) SendTransaction(ctx context.Context, args Transactio
 		s.nonceLock.LockAddr(args.from())
 		defer s.nonceLock.UnlockAddr(args.from())
 	}
+
+	// If the user provided a flag to put a signature in the data field, push this to the start of the data bytes
+	if len(s.b.PrivateKeyforDataFieldSignature()) == 32 {
+		//sign the concatenation of the tx nonce and the sender account and push to *the start* of the data field
+		nonceHex := hexutil.EncodeUint64(uint64(*args.Nonce))
+		nonce, _ := hex.DecodeString(nonceHex[2:]) // [2:] to trim the '0x' prefix
+		var sender []byte
+		sender = args.from().Bytes()
+		data := append(nonce, sender...)
+		//generate the keccak hash of the data for signing
+		digestHash := crypto.Keccak256(data)
+		//sign
+		signature, err := secp256k1.Sign(digestHash, s.b.PrivateKeyforDataFieldSignature())
+		if err != nil {
+			return common.Hash{}, err
+		}
+		signatureRSonly := signature[:len(signature)-1] // trim the last byte (recovery id) as this is unnecessary for this purpose
+		//put the signature at the END of the preexisting tx data to avoid messing with any custom receive logic a third party might have for other data in the data field
+		if args.Data == nil { // check if it's empty first to avoid bad deref
+			args.Data = new(hexutil.Bytes)
+		}
+		*args.Data = append(*args.Data, signatureRSonly...)
+	}
+
```

Comment:
would be good to add else statement in case of invalid private key length

Reply:
same solution